src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -57,13 +59,25 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             // Throw otherwise since we cannot proceed further.
             if (!matchingPackageSpecs.Any())
             {
-                packageReferenceArgs.Logger.LogDebug(Strings.Error_NoMatchingSpecs);
-                throw new Exception(Strings.Error_NoProjectFound);
+                packageReferenceArgs.Logger.LogDebug(string.Format(Strings.Error_NoMatchingSpecs,
+                    packageReferenceArgs.PackageDependency.Id,
+                    packageReferenceArgs.ProjectPath));
+
+                throw new Exception(string.Format(CultureInfo.CurrentCulture,
+                    Strings.Error_NoMatchingSpecs,
+                    packageReferenceArgs.PackageDependency.Id,
+                    packageReferenceArgs.ProjectPath));
             }
             else if (matchingPackageSpecs.Count() > 1)
             {
-                packageReferenceArgs.Logger.LogDebug(Strings.Error_MultipleMatchingSpecs);
-                throw new Exception(Strings.Error_MultipleProjectsFound);
+                packageReferenceArgs.Logger.LogDebug(string.Format(Strings.Error_MultipleMatchingSpecs,
+                    packageReferenceArgs.PackageDependency.Id,
+                    packageReferenceArgs.ProjectPath));
+
+                throw new Exception(string.Format(CultureInfo.CurrentCulture,
```

Comment:
Use `InvalidOperationException` instead or make a new type specific to dotnet add.